File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {

        Map<String, List<String>> map = new HashMap<>();
        List<List<String>> result = new ArrayList<>();

        for(int i=0; i<strs.length; i++) {
            char[] ch = strs[i].toCharArray();
            Arrays.sort(ch);

            String s = new String(ch);

            if(!map.containsKey(s)) {
                List<String> tempList = new ArrayList<>();
                tempList.add(strs[i]);
                map.put(s, tempList);
            } else {
                List<String> tempList = map.get(s);
                tempList.add(strs[i]);
                map.put(s, tempList);
            }
        }

        Set<String> keys = map.keySet();

        for(String key : keys) {
            result.add(map.get(key));
        }

        return result;
    }
}

File: 49-group-anagrams/group-anagrams.cs
public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        Dictionary<string, IList<string>> strMap = new Dictionary<string, IList<string>>();
        List<IList<string>> result = new List<IList<string>>();
        for(int i=0; i<strs.Length; i++) {
            char[] ch = strs[i].ToCharArray();

            Array.Sort(ch);

            string s = new string(ch);

            if(!strMap.ContainsKey(s)) {
                List<string> temp = new List<string>();
                temp.Add(strs[i]);
                strMap[s] = temp;
            } else {
                IList<string> temp = strMap[s];
                temp.Add(strs[i]);
                strMap[s] = temp;
            }
        }

        foreach(var val in strMap) {
            result.Add(val.Value);
        }

        return result;
    }
}